lib/Db/MessageMapper.php

```diff
@@ -430,6 +407,39 @@ public function updateBulk(Account $account, Message ...$messages): array {
 		return $messages;
 	}
 
+	private function updateTags(Account $account, Message $message, array $tags, PerformanceLoggerTask $perf): void {
+		$imapTags = $message->getTags();
+		$dbTags = $tags[$message->getMessageId()] ?? [];
+
+		if (empty($imapTags) === true && empty($dbTags) === true) {
+			// neither old nor new tags
+			return;
+		}
+
+		$toAdd = array_udiff($imapTags, $dbTags, static function (Tag $a, Tag $b) {
+			return strcmp($a->getImapLabel(), $b->getImapLabel());
+		});
+		foreach ($toAdd as $tag) {
+			// add logging
```

Comment:
logging?

Reply:
left over from previous commits, removed it